src/SFA.DAS.EmployerIncentives.Commands/Persistence/IAccountDomainRepository.cs

```diff
@@ -1,9 +1,13 @@
 ﻿using SFA.DAS.EmployerIncentives.Abstractions.Domain;
 using SFA.DAS.EmployerIncentives.Domain.Accounts;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.EmployerIncentives.Commands.Persistence
 {
     public interface IAccountDomainRepository : IDomainRepository<long, Account>
     {
+        Task<IEnumerable<Account>> GetByLegalEntityId(long legalEntityId);
```

Comment:
These shouldn't be on the domain respository.  Use the IQueryRepository to get data then look up the domain entity using the IDomainRepository.Find (by entity Id).

The entity Id is the key to the domain model.  It's the logical write model it's not designed for querying.

Reply:
For what benefit?  It'll slow down the process and bloat the code.

Reply:
Because this interface is just for hydrating and persisting a single domain object from the logical write store.

There is already an existing IQueryRepository to retrieve data so that should be used as that is it's concern then publish the commands to do the work on each individual domain object.  That would be more performant since it will could handle multiple domain objects in parallel each observing he existing locking strategy.

By adding a query to this interface encourages batch processing.